jscomp/syntax/src/res_core.ml

```diff
@@ -3923,6 +3941,30 @@ and parseListExpr ~startPos p =
             loc))
       [(Asttypes.Nolabel, Ast_helper.Exp.array ~loc listExprs)]
 
+and parseDictExpr ~startPos p =
+  let exprs =
+    parseCommaDelimitedRegion ~grammar:Grammar.DictRows ~closing:Rbrace
+      ~f:parseDictExprRow p
+  in
+  let loc = mkLoc startPos p.prevEndPos in
+  let toKeyValuePair recordItem =
+    match recordItem with
+    | {Location.txt = Longident.Lident key}, valueExpr ->
+      Some
+        (Ast_helper.Exp.tuple ~loc
```

Comment:
Not sure if I'm reading the code correct, but is the `~loc` here being set to the location of all dict rows combined? I think we want the location of every tuple to cover a specific pair: `"key": "value"` .